examples/download_everything.py

```diff
@@ -0,0 +1,272 @@
+                json.dump(a.params.to_dict_full(), f, indent=4)
+
+            preconditions_path = os.path.join(assert_path, "preconditions")
+            os.makedirs(preconditions_path)
+
+            for precondition in a.preconditions()[0]:
+                precondition_path = os.path.join(
+                    preconditions_path,
+                    str(precondition.params.assert_precondition_id),
+                )
+                os.makedirs(precondition_path)
+
+                with open(
+                    os.path.join(precondition_path, "precondition.json"), "w"
+                ) as f:
+                    json.dump(precondition.params.to_dict_full(), f, indent=4)
+
+        groups_path = os.path.join(test_path, "groups")
+        os.makedirs(groups_path)
+
+        for group in test.groups()[0]:
+            group_path = os.path.join(groups_path, str(group.params.group_id))
+            os.makedirs(group_path)
+
+            with open(os.path.join(group_path, "group.json"), "w") as f:
+                json.dump(group.params.to_dict_full(), f, indent=4)
+
+            participants_path = os.path.join(group_path, "participants")
+            os.makedirs(participants_path)
+
+            for participant in group.participants()[0]:
+                participant_path = os.path.join(
+                    participants_path, str(participant.params.participant_id)
+                )
+                os.makedirs(participant_path)
+
+                with open(
+                    os.path.join(participant_path, "participant.json"), "w"
+                ) as f:
+                    json.dump(participant.params.to_dict_full(), f, indent=4)
+
+        participants_path = os.path.join(test_path, "participants")
+        os.makedirs(participants_path)
+
+        for participant in test.participants(
+            query_params=QueryParams().filter(
+                ParticipantFilterKey.HAS_GROUP, "false"
+            )
+        )[0]:
+
+            participant_path = os.path.join(
+                participants_path, str(participant.params.participant_id)
+            )
+            os.makedirs(participant_path)
+
+            with open(
+                os.path.join(participant_path, "participant.json"), "w"
+            ) as f:
+                json.dump(participant.params.to_dict_full(), f, indent=4)
+
+
+arg_parser = argparse.ArgumentParser(
+    prog="project dump",
+    description="download project data",
+)
+
+arg_parser.add_argument("--project_id", type=int, required=True)
+arg_parser.add_argument("--access_token", type=str, required=True)
+arg_parser.add_argument("--api_base", type=str, required=False)
+arg_parser.add_argument("--output_dir", type=str, required=True)
+arg_parser.add_argument(
+    "--limit",
+    type=int,
+    default=1,
+    help="limits the number of test and run resources to download. "
+    "order_by=-id is used to get the latest resources. "
+    "set to 0 to download all resources",
```

Comment:
this will very definitely be terrible. either the script will be mega slow, or it will destroy our infra. i think more reasonable idea would be to download a specific test or specific run. the motivation for the script is actually to download data for a specific test run